src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -199,13 +243,141 @@ public override void OnRemoveBehavior()
         try
         {
             var res = (await _crpgClient.UpdateUsersAsync(new CrpgGameUsersUpdateRequest { Updates = userUpdates })).Data!;
-            SendRewardToPeers(res.UpdateResults, crpgPeerByUserId, valorousPlayerIds);
+            // elements from netCompensationByCrpgUserId will be removed in SendRewardToPeers as soon as the Messages were sent
+            SendRewardToPeers(res.UpdateResults, crpgPeerByUserId, valorousPlayerIds, netCompensationByCrpgUserId);
+            // apply compensation for disconnected users
+            await _crpgClient.UpdateUsersAsync(new CrpgGameUsersUpdateRequest { Updates = GetCompensationUpdatesForUsers(netCompensationByCrpgUserId) });

```

Comment:
Weren't the rewards already sent to disconnected players?

Reply:
Done
Reworked the code a bit to ensure only the compensation gets sent to disconnected players
Problem was, if you disconnect you would still get the full reward for the round and i think that probably would not be good.